### PR TITLE
chore(core): refresh API docs for 17.0

### LINKED ISSUE #10580: docs(core): v17 APIs
1. New APIs
2. Removed APIs
3. Moving from v16 to v17 -- in particular context and action interfaces
4. Future direction as needed

---

## Copied from keymanapp/help.keyman.com#909

Guide updates:
- Upgrading from 16.0 to 17.0
- Document how to use new context + action work (ref https://github.com/keymanapp/keyman/issues/9999)

Reference updates:
- renamed functions, structures, files
- km_core_state_context_set_if_needed
- km_core_actions and friends
- km_core_context_items_to_utf32
- km_core_state_get_actions
- km_core_actions_dispose
- km_core_state_context_clear
- km_core_event_flags

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10580.

@keymanapp-test-bot skip

### CODE REVIEW

File: core/include/keyman/keyman_core_api.h [hunk 1]
```diff
@@ -435,215 +954,419 @@ km_core_state_options_to_json(km_core_state const *state,
...
+## Parameters
+
+`state`
+: A pointer to the opaque state object
+
+`imx_callback`
+: pointer to a function that implements the IMX callback
+
+`callback_object`
+: TODO
```
Comment (rc-swag) [inline core/include/keyman/keyman_core_api.h:1335]:
```suggestion
: An opaque pointer that can be used to pass context information to the callback function, 
  usually it is a user-defined data structure.
```

**Review by rc-swag — COMMENTED** [review]
two suggestions.

**Review by rc-swag — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.293-beta](https://keyman.com/downloads/releases/beta/17.0.293)

**Pull request merged**